### PR TITLE
initial commit for iter() docs

### LINKED ISSUE #1167: [Term Entry] Python Built-in Functions: `iter()`
### Reason/inspiration (optional)

We'd like someone to write a new term entry about the built-in `iter()` function in Python. Be sure to include the following information:

- A "Syntax" section with a \```pseudo code block.
- An "Example" section with a \```py code block.
- A "Codebyte Example" section with a \```codebyte/python block.

A nice-to-have:

- Link out to the [existing entry](https://www.codecademy.com/resources/docs/python/iterators) about Iterators in Python.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by mcuocolo**
### Description

A contribution proposal to the Codeacademy Docs for the iter() method. I used personal knowledge and information gathered online mainly from python official documentation.
This is my firts pull request ever. This is to complete the full course related to git/github on Codeacademy.

Closes #1167

### Type of Change

- Adding a new entry

### Checklist

- [x ] All writings are my own.
- [ x] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by yangc95** [comment]
> All testing using yarn yields the same result : 
> 
> command failed with exit code 1
> 
> 

Did you install the dependencies first?

File: content/python/concepts/iterators/terms/iter.md
Comment (SSwiniarski) [inline content/python/concepts/iterators/terms/iter.md:23]: This would be better as a bulleted list than a table.

File: content/python/concepts/iterators/terms/iter.md
Comment (SSwiniarski) [inline content/python/concepts/iterators/terms/iter.md:58]:
```suggestion
When using iterables, it is usually not necessary to call `iter()` or deal with iterator objects. The for statement does that automatically, creating a temporary unnamed variable to hold the iterator for the duration of the loop.
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@mcuocolo, I did a first review and left a couple of comments.

File: content/python/concepts/iterators/terms/iter.md
Comment (mcuocolo) [reply content/python/concepts/iterators/terms/iter.md:23]: I'll change that.
(... was so happy to make markdown table ;-) )

Comment (mcuocolo) [reply content/python/concepts/iterators/terms/iter.md]: I added the requested bullet list

File: content/python/concepts/iterators/terms/iter.md
Comment (SSwiniarski) [inline content/python/concepts/iterators/terms/iter.md:15]:
```suggestion
The **`iter()`** method is a python built-in function that returns an iterator object for a given `object`.
```

**Review by SSwiniarski — APPROVED** [review]
@mcuocolo one more comment, otherwise it's ready for a second review.

File: content/python/concepts/iterators/terms/iter.md
Comment (mcuocolo) [reply content/python/concepts/iterators/terms/iter.md]: @SSwiniarski  updated as requested adding bold for iter()

... must admit I'm a little lost with branch and github.
Everything seemed so easy in the tutorial ...

Comment (SSwiniarski) [reply content/python/concepts/iterators/terms/iter.md]: @mcuocolo, one of the Codecademy team will come by this entry and give a second review. You don't have to do anything.

File: content/python/concepts/iterators/terms/iter.md
Comment (Dusch4593) [inline content/python/concepts/iterators/terms/iter.md:52]:
```suggestion
class codeNinjaCount:
  def __init__(self, greeting, num):
    self.greeting = greeting
    self.num = num
    print(f"{self.greeting}! Let's count:")
  
  def __call__(self):
    self.num += 1
    return self.num

codeNinjaIterable = iter(codeNinjaCount("Hello, World", 1), 10)

while True:
  try:
    print(next(codeNinjaIterable))
  except StopIteration:
    print("StopIteration Exception raised. Value returned was equal to sentinel value")
    break
```

The example you provided is almost exactly like the one given in the [official Python docs on `iter()`](https://docs.python.org/3/library/functions.html#iter). I'd suggest using a Python class to demonstrate the `sentinel` parameter.

File: content/python/concepts/iterators/terms/iter.md
Comment (Dusch4593) [inline content/python/concepts/iterators/terms/iter.md:54]:
```suggestion

This will produce the following output:

```

And then let's add a \```shell code block below that looks like this:

```shell
Hello, World! Let's count:
2
3
4
5
6
7
8
9
StopIteration Exception raised. Value returned was equal to sentinel value
```

File: content/python/concepts/iterators/terms/iter.md
Comment (Dusch4593) [inline content/python/concepts/iterators/terms/iter.md:55]:
```suggestion
### Using iter() with iterables
```

File: content/python/concepts/iterators/terms/iter.md
Comment (Dusch4593) [inline content/python/concepts/iterators/terms/iter.md:57]:
```suggestion
When using iterables, it usually isn't necessary to call `iter()` or deal with iterator objects. The `for` loop does this automatically by creating a temporary, unnamed variable to hold the iterator for the duration of the loop.
```

File: content/python/concepts/iterators/terms/iter.md
Comment (Dusch4593) [inline content/python/concepts/iterators/terms/iter.md:59]:
```suggestion
The following code examples use the `iter()` function and a `for` loop to produce the same results:
```

File: content/python/concepts/iterators/terms/iter.md
Comment (Dusch4593) [inline content/python/concepts/iterators/terms/iter.md:73]:
```suggestion
for fruit in fruits:
  print(fruit)
```

File: content/python/concepts/iterators/terms/iter.md
Comment (Dusch4593) [inline content/python/concepts/iterators/terms/iter.md:47]:
```suggestion
The following example applies the optional `sentinel` parameter through a callable class object (though the `__call__()` method:

```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@mcuocolo Finished with my review and left some comments/suggestions 😄

**Comment by Dusch4593** [comment]
Hey @mcuocolo did you have any questions or need any help with implementing the latest round of feedback? As a reminder, today is the last day to merge your PRs and have them count toward Hacktoberfest. 👀

File: content/python/concepts/iterators/terms/iter.md
Comment (mcuocolo) [reply content/python/concepts/iterators/terms/iter.md:55]: I don't understand this request.

Comment (Dusch4593) [reply content/python/concepts/iterators/terms/iter.md:80]: @mcuocolo See the changes I applied [here](a66477c094d61e22195b61b43b777487f28fef52) to see what I mean.

**Review by Dusch4593 — APPROVED** [review]
@mcuocolo This is looking ready for merge! 😄

**Comment by mcuocolo** [comment]
I'm sorry to tell you that I don't even know how to edit the file anymore.

I commit/accepted all suggestions, but :
 - My local file is outdated and branch is not syncing.
 - Can't find the file to edit.
 - I know now that working with github is 1000 times worse than track change in word and I honestly thaught that it was impossible.
 - I'd like to help, but I must admit I'm way over my head with my beginner git skills.
I honestly tried ...

**Comment by Dusch4593** [comment]
@mcuocolo You're totally fine! 😄

That last suggestion of mine was pretty intense, honestly. I went ahead and updated in a66477c094d61e22195b61b43b777487f28fef52.

**Super important:** In 4f45ed0843f7486270e3c4abd9de3dc9363e5d82, I also nested the `iter.md` file in a similarly-named `iter` folder in order for the GitHub checks to pass. This is probably why you're running into trouble on your local machine.

Git and GitHub were super challenging for me to get used to. I definitely sympathize with your struggle and definitely encourage you to practice these skills with the following:

- Auditing/completing our [Learn Git & GitHub course](https://www.codecademy.com/learn/learn-git) with Codecademy.
- Continuing to apply this skill by contributing to Codecademy Docs (and other stuff on GitHub)!

**Comment by mcuocolo** [comment]
Hello @Dusch4593 ,
Thanks for your support. 
However, the merged commit doesn't seem right. There's no logic between examples and explanation.
I tried my best...

**Comment by Dusch4593** [comment]
@mcuocolo All we can do is our best 😄 Thank you again for your contribution to Codecademy Docs!

The entry should now be live: https://www.codecademy.com/resources/docs/python/iterators/iter

> However, the merged commit doesn't seem right. There's no logic between examples and explanation.
I agree with you that there seems to be some inconsistencies within the examples/explanations. I was wondering if you'd be open to opening a new PR to make some edits to this entry? Let me know if this interests you. 😄

My suggested changes would include the following:

- Let's move the this term entry to under the "Built-in Functions" concept:
  - ~`python/concepts/iterators/terms/iter/iter.md`~ --> `python/concepts/built-in-functions/terms/iter/iter.md`
- On line # 3, in the `Description`, let's change the text to say `'Returns an iterator object.'`

If you open a new PR, I'll add more feedback to specify how the \```shell output snippets should be rearranged.

**Comment by mcuocolo** [comment]
OK to continue improving.
So directly from github I created a new PR : `mcuocolo:patch-1`
... and waiting for the suggestions to add all corrections at once.
However, I don't know how to move the file to new dir without forking the whole project.

**Comment by yangc95** [comment]
@mcuocolo Hey! I'm slightly confused by this inquiry. I do see your PR [1206](https://github.com/Codecademy/docs/pull/1206).

Looks like there is a comment on there by one maintainer. Once that comment gets resolved and the maintainer approves, it will require another review (or more suggestions) by another maintainer.

By moving the file to a new directory or folder, could you provide images of where you are stuck?

**Pull request merged**
